### PR TITLE
accelerate merge clusters during schedule and improve test coverage

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
reduce unnecessary code operations in some cases and improve test coverage
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/binding.go [hunk 1]
```diff
@@ -87,6 +87,12 @@ func DivideReplicasByTargetCluster(clusters []workv1alpha2.TargetCluster, sum in
 
 // MergeTargetClusters will merge the replicas in two TargetCluster
 func MergeTargetClusters(old, new []workv1alpha2.TargetCluster) []workv1alpha2.TargetCluster {
+	switch {
```
Comment (XiShanYongYe-Chang) [inline pkg/util/binding.go:90]: IMOP, this logic may seem a little redundant, after all, the following logic already overrides this judgment.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Other parts LGTM

File: pkg/util/binding.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/util/binding.go:90]: This is the short path.
If we need a short patch depends on :
1. How often does the short patch hit? 
2. The following logic is complicated and most cases don't need to be involved.

cc @Garrybest for comment

Comment (Garrybest) [reply pkg/util/binding.go:90]: When first scheduled, the `old` cluster must be empty. So I think this patch is useful.

**Comment by RainbowMango** [comment]
/retitle accelerate merge clusters during schedule and improve test coverage
/lgtm
/approve

File: pkg/util/binding.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/util/binding.go:90]: ok. here we go.

**Pull request merged**